### ISSUE TITLE
PyDeadObjectError('The C++ part of the ActivityPanel object has been deleted, attribute access no longer allowed.',)

### ISSUE THREAD

**Issue by brussee** [Tribler/tribler#2436]
```
======================================================================
ERROR: Test whether adding a torrent def to a channel works
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/test_as_server.py", line 282, in tearDown
    super(TestAsServer, self).tearDown(annotate=False)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 157, in g
    d = func(*args, **kargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/test_as_server.py", line 166, in tearDown
    process_unhandled_exceptions()
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/util.py", line 116, in check_exceptions
    raise Exception("Test raised %d unhandled exceptions, last one was: %s" % (exc_counter, last_exc))
Exception: Test raised 5 unhandled exceptions, last one was: PyDeadObjectError('The C++ part of the ActivityPanel object has been deleted, attribute access no longer allowed.',)
```
https://jenkins.tribler.org/job/GH_Tribler_PR_tests_linux/2516/artifact/output/test_runners_output/00009.err/*view*/

**Comment by whirm** [comment]
It's due to the observer added (self.on_tunnel_remove) not being removed when the gui is destroyed. It's being removed later when the community get's unloaded but it's already too late.

I've pushed a quick hack to work around it. As the gui is going away I guess it's not worth fixing it properly.

**Comment by whirm** [comment]
The commit will be merged to next with @brussee 's #2423
